### PR TITLE
SOAR 272: Added Workflow URLScan with Slack

### PR DESCRIPTION
**Pull request by tylerslijboom-r7**
## Requirements

Added workflow URLScan with Slack

### Checklist
- [x] Workflow verified as working in a test
- [x] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [x] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Workflow includes a detailed Description
- [x] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [x] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [x] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [x] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

Post your results with screenshots in this PR using the following style:

### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

#### Workflow Builder

workflow1

workflow2

#### Job

job1

job2

#### Artifact

artifact1

### CODE REVIEW

File: workflows/URLScan_with_Slack/URLScan_with_Slack.icon [hunk 1]
```diff
@@ -0,0 +1,1345 @@
...
+  "kom": {
+    "komandVersion": "1.49.0-0-gc013d",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-04-20T14:20:02.994596322Z",
+    "workflowVersions": [
+      {
+        "name": "URLScan with Slack",
+        "type": "runnable",
+        "version": "",
+        "description": "Report back the full report of a URL in Slack, using the urlscan.io plugin",
```
Comment (mrinehart-r7) [inline workflows/URLScan_with_Slack/URLScan_with_Slack.icon:11]:
```suggestion
        "description": "Retrieve the full URLScan.io report of a URL in Slack",
```

File: workflows/URLScan_with_Slack/help.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+* [Slack](https://insightconnect.help.rapid7.com/docs/configure-slack-for-chatops)
+* URLScan.io API Key
+
+# Documentation
+
+## Setup
+
+Once the workflow has been imported, edit the workflow and setup or select your Slack connection in the _URLScan Slack Trigger_ step.
+
+The URLScan connection needs to be setup for steps _Submit URL for Scan_ and _Get URLScan Report_
```
Comment (mrinehart-r7) [inline workflows/URLScan_with_Slack/help.md:24]:
```suggestion
The URLScan connection needs to be setup for steps _Submit URL for Scan_ and _Get URLScan Report_.
```

File: workflows/URLScan_with_Slack/workflow.spec.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    title: Workflow View, Top section
+  - name: workflow2.png
+    title: Workflow View, Bottom section
+  - name: slack1.png
+    title: Completed Job In Slack with Valid Input
+  - name: slack2.png
+    title: Completed Job In Slack with Blacklisted Domain
+  - name: slack3.png
+    title: Completed Job In Slack with Invalid Input
+  - name: artifact.png
```
Comment (mrinehart-r7) [inline workflows/URLScan_with_Slack/workflow.spec.yaml:32]:
```suggestion
  - name: artifact1.png
```

File: workflows/URLScan_with_Slack/URLScan_with_Slack.icon [hunk 4]
```diff
@@ -0,0 +1,1345 @@
...
+          "7b8d2f66-7d80-4002-bb1a-75e272612681": {
+            "nodeId": "7b8d2f66-7d80-4002-bb1a-75e272612681",
+            "name": "Print Report Submission",
+            "type": "action_chatops",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "attachments": [],
+                "message": "`<http://rapid7.com|{{[6f80dc1f-9a48-4fbd-bfa0-06f2a86a6c04].[content]}}>` submitted",
```
Comment (jmcadams-r7) [inline workflows/URLScan_with_Slack/URLScan_with_Slack.icon:771]: Give the user a message here like "Waiting for scan" or "Sleeping for 30 seconds to wait for scan results." Something to indicate we're about to have a big delay.

File: workflows/URLScan_with_Slack/help.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
+# Description
+
+This workflow accepts a Slack command containing a URL, looks up the URL with urlscan.io service and posts the scan report to Slack.
```
Comment (mrinehart-r7) [inline workflows/URLScan_with_Slack/help.md:3]:
```suggestion
This workflow accepts a Slack command containing a URL, looks up the URL with urlscan.io service, and posts the scan report to Slack.
```

File: workflows/URLScan_with_Slack/workflow.spec.yaml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+extension: workflow
+products: ["insightconnect"]
+name: URLScan_with_Slack
+title: "URLScan with Slack"
+description: This workflow accepts a Slack command, looks up the URL in the urlscan.io service and prints the scan report back to slack
```
Comment (mrinehart-r7) [inline workflows/URLScan_with_Slack/workflow.spec.yaml:5]:
```suggestion
description: This workflow accepts a Slack command, looks up the URL in the urlscan.io service, and prints the scan report back to slack
```

File: workflows/URLScan_with_Slack/workflow.spec.yaml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+name: URLScan_with_Slack
+title: "URLScan with Slack"
+description: This workflow looks up a URL given via Slack command in urlscan.io and reports information on the URL back to Slack
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [threat_detection_and_response]
+  keywords: [threat_detection_and_response, chat, chatops, phishing, work_from_home, slack, covid]
```
Comment (jschipp-r7) [inline workflows/URLScan_with_Slack/workflow.spec.yaml:12]: We shouldn't duplicate use_cases and keywords

```suggestion
  keywords: [chat, chatops, phishing, work_from_home, slack, covid]
```

**Pull request merged**